### PR TITLE
[stable10] Do not display technical error messages

### PR DESCRIPTION
**Pull request by LukasReschke**
Backport of https://github.com/nextcloud/gallery/pull/154

No backport to older versions required.

cc @MorrisJobke @oparoz Please review.

### CODE REVIEW

**Comment by oparoz** [comment]
I assume `createMock` was introduced in NC11?

**Comment by MorrisJobke** [comment]
> I assume createMock was introduced in NC11?

It's in phpunit 5.x and we added a backport for phpunit 4.8: https://github.com/nextcloud/server/pull/1396/files#diff-c24f6db53e742461694bff7f95900daa

**Comment by MorrisJobke** [comment]
Tested and works 👍

**Comment by oparoz** [comment]
>It's in phpunit 5.x and we added a backport for phpunit 4.8: https://github.com/nextcloud/server/pull/1396/files#diff-c24f6db53e742461694bff7f95900daa

OK, but it doesn't seem to be working properly. Any idea why?

File: tests/unit/controller/HttpErrorTest.php [hunk 1]
```diff
@@ -72,12 +74,32 @@ public function providesExceptionData() {
 	 * @param String $status
 	 */
 	public function testJsonError($exception, $message, $status) {
-		$httpError = $this->getMockForTrait('\OCA\Gallery\Controller\HttpError');
+		$request = $this->createMock(IRequest::class);
```
Comment (MorrisJobke) [inline tests/unit/controller/HttpErrorTest.php]: This is PHP 5.6+ only ... use the string instead ;)

**Comment by MorrisJobke** [comment]
> OK, but it doesn't seem to be working properly. Any idea why?

I fixed the `::class` stuff that is 5.6+ only

File: tests/unit/controller/HttpErrorTest.php (hunk 1 above)
Comment (oparoz) [inline tests/unit/controller/HttpErrorTest.php]: OK, but it's failing on all versions, so there must be something else. I'll try locally ASAP.

**Comment by oparoz** [comment]
>I fixed the ::class stuff that is 5.6+ only

Thanks :)

**Comment by MorrisJobke** [comment]
And I now also added the wrapper. This was not backported to stable10.

**Comment by oparoz** [comment]
>And I now also added the wrapper. This was not backported to stable10.

Ah ha! Mystery solved! Thanks @MorrisJobke :)

**Comment by oparoz** [comment]
Don't merge right away as I'm going to move that method to a parent class to make it available to other tests in case we need to use it again somewhere else.

**Comment by MorrisJobke** [comment]
👍 

@oparoz Is this okay to merge once CI is fine?

**Comment by oparoz** [comment]
Yes!

**Pull request merged**